src/state/reducers/minimumVersion.ts

```diff
@@ -0,0 +1,77 @@
+import { Reducer } from 'redux-automaton';
+import * as RD from '@devexperts/remote-data-ts';
+import * as automaton from 'redux-automaton';
+import * as actions from '../actions';
+
+import * as minimumVersionApi from '../../api/minimumVersion';
+
+import { Action } from '../actions';
+import { AppState } from '../types';
+import { withToken } from './accessToken';
+
+import { pipe } from 'fp-ts/lib/function';
+
+export const initialState = RD.initial;
+
+export const reducer: Reducer<AppState['minimumVersion'], Action> = (
+  state = initialState,
+  action,
+) => {
+  switch (action.type) {
+    case 'minimumVersion/get/start': {
+      if (RD.isPending(state)) {
+        return state;
+      }
+
+      return automaton.loop(
+        RD.pending,
+        withToken(
+          minimumVersionApi.fetchVersions,
+          actions.make('minimumVersion/get/end'),
+        ),
+      );
+    }
+
+    case 'minimumVersion/get/end': {
+      return pipe(action.payload, RD.fromEither);
+    }
+
+    default: {
+      return state;
+    }
+  }
+};
+
+const isVersionBigEnough = (
+  clientVersion: minimumVersionApi.AppClient,
+  requiredVersion: minimumVersionApi.AppClient,
+) => {
+  if (clientVersion.major > requiredVersion.major) return true; // eslint-disable-next-line padding-line-between-statements
+  if (clientVersion.major < requiredVersion.major) return false;
+
+  // if major version equal, cmpare minor
+  if (clientVersion.minor > requiredVersion.minor) return true; // eslint-disable-next-line padding-line-between-statements
+  if (clientVersion.minor < requiredVersion.minor) return false;
+
+  // compare patch
+  return clientVersion.patch >= requiredVersion.patch;
+};
```

Comment:
Add return type to fn;

This is tooo complex this should be just
```
return clientVersion.major >= requiredVersion.major
&& clientVersion.minor >= requiredVersion.minor && clientVersionPatch >= requiredVersionPatch
```

Reply:
it wont work like this.
What if the version is 3.0.1 and minimum required is 2.19.0?

Reply:
Imo this function is not complex at all. 
1. check major
2. if equal, check minor
3. if equal check patch